packages/grid/x-grid-modules/src/hooks/features/useColumnReorder.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { ColDef } from '../../models/colDef';
+import { useLogger } from '../utils';
+import { ApiRef } from '../../models';
+
+export const useColumnReorder = (
+  apiRef: ApiRef
+) => {
+  const logger = useLogger('useColumnReorder');
+
+  const dragCol = React.useRef<ColDef | null>();
+  const dragColNode = React.useRef<HTMLElement | null>();
+
+  const handleDragStart = React.useCallback(
+    (col: ColDef, htmlEl: any): void => {
+      logger.debug(`Start dragging col ${col.field}`);
```

Comment:
Here we should emit an event that users can hook into. 
`onColumnDragStart` handler should be triggered.
Have a look at how `options.onRowSelected` works.

We should also add the api method to register this handler
`useSelection` can be a reference